src/randomizer.py

```diff
@@ -1447,6 +1448,10 @@ def perm_add_mask(
     mask = random.choice(masks) + random.choice(["", "u"])
 
     new_expr = ca.BinaryOp("&", expr, ca.Constant("int", mask))
+    if random_bool(random, 0.5):
```

Comment:
0.5 feels a tad high to me for something which is certainly a fakematch, but at the same time it's just a 1/12 chance the number of bitands will be correct... hm. Maybe lower to 0.3 and also raise perm_add_mask weight 10->15?